lib/Controller/Library.php

```diff
@@ -2616,16 +2621,39 @@ public function connectorImport(Request $request, Response $response)
         }
 
         // Process all of those downloads
-        $this->mediaFactory->processDownloads(null, function ($media) use ($importQueue) {
-            // Failure
-            // Pull out the import which failed.
-            foreach ($importQueue as $import) {
-                /** @var ProviderImport $import */
-                if ($import->media->getId() === $media->getId()) {
-                    $import->setError(__('Download failed'));
+        $this->mediaFactory->processDownloads(
+            function ($media) use ($importQueue, $libraryLocation) {
+                // Success
+                // if we have pictureId (pixabay) get the thumbnail for the video and save it to library
+                foreach ($importQueue as $import) {
+                    /** @var ProviderImport $import */
+                    if ($import->media->getId() === $media->getId() && $media->mediaType === 'video' && !empty($import->searchResult->pictureId)) {
+                        try {
+                            $client = new Client();
+                            $client->request(
+                                'GET',
+                                str_replace('pictureId', $import->searchResult->pictureId, 'https://i.vimeocdn.com/video/pictureId_960x540.png'),
```

Comment:
Could we do this string replace inside the search results instead? that way we could in theory show a thumbnail in the UI if we wanted to?